src/components/Table.tsx

```diff
@@ -144,16 +204,28 @@ export default class Table<T> extends React.Component<
 	}
 
 	sortData(data: T[]): T[] {
-		if (this.state.sortColumn === null) {
+		const { sortColumn } = this.state;
+		if (sortColumn === null) {
 			return data;
 		}
+		const column =  find(this.props.columns, { field: sortColumn });
 
-		const collection = sortBy<T>(data.slice(), item => {
-			const sortableValue = item[this.state.sortColumn!] as any;
-			return isPlainObject(sortableValue)
-				? (sortableValue as any).value
-				: sortableValue;
-		});
+		if (!column) {
+			return data;
+		}
+
+		let collection;
+
+		if (isFunction(column.sortable)) {
+			collection = data.slice().sort(column.sortable);
```

Comment:
Is this new?

Reply:
This is intended behaviour that wasn't implemented. I only discovered that providing a sort function doesn't work because I wrote a test for it.